public/app/features/plugins/sql/components/TableSelector.tsx

```diff
@@ -4,32 +4,27 @@ import { useAsync } from 'react-use';
 import { SelectableValue, toOption } from '@grafana/data';
 import { Select } from '@grafana/ui';
 
-import { QueryWithDefaults } from '../defaults';
 import { DB, ResourceSelectorProps } from '../types';
 
 interface TableSelectorProps extends ResourceSelectorProps {
   db: DB;
-  value: string | null;
-  query: QueryWithDefaults;
+  table: string | null;
+  dataset: string | undefined;
   onChange: (v: SelectableValue) => void;
-  forceFetch?: boolean;
 }
 
-export const TableSelector = ({ db, query, value, className, onChange, forceFetch }: TableSelectorProps) => {
+export const TableSelector = ({ db, dataset, table, className, onChange }: TableSelectorProps) => {
   const state = useAsync(async () => {
-    if (!query.dataset && !forceFetch) {
-      return [];
-    }
-    const tables = await db.tables(query.dataset);
+    const tables = await db.tables(dataset);
```

Comment:
I wouldn't remove the condition from here as it prevents it to run when dataset is undefined.

Reply:
Ah yes, good point. I'll add back the logic to deal with that.